io/eolearn/tests/test_local_io.py

```diff
@@ -0,0 +1,266 @@
+
+        eop = EOPatch.load(self.PATCH_FILENAME)
+        eop.mask_timeless['mask_timeless'] = mask_timeless
+
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            tmp_file_name = 'temp_file.tiff'
+            task = ExportToTiff((FeatureType.MASK_TIMELESS, 'mask_timeless'), folder=tmp_dir_name)
+            task.execute(eop, filename=tmp_file_name)
+
+            raster = read_data(os.path.join(tmp_dir_name, tmp_file_name))
+            self.assertTrue(np.all(subset == raster))
+
+    def test_export2tiff_mask_single(self):
+        mask = np.arange(5*3*3*1).reshape(5, 3, 3, 1)
+        times = [4]
+        subset = mask[times].squeeze()
+
+        eop = EOPatch.load(self.PATCH_FILENAME)
+        eop.mask['mask'] = mask
+
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            tmp_file_name = 'temp_file.tiff'
+            task = ExportToTiff((FeatureType.MASK, 'mask'), folder=tmp_dir_name, date_indices=times)
+            task.execute(eop, filename=tmp_file_name)
+
+            raster = read_data(os.path.join(tmp_dir_name, tmp_file_name))
+            self.assertTrue(np.all(subset == raster))
+
+    def test_export2tiff_mask_list(self):
+        mask = np.arange(5*3*2*1).reshape(5, 3, 2, 1)
+        times = [4, 2]
+        subset = mask[times].squeeze()
+
+        eop = EOPatch.load(self.PATCH_FILENAME)
+        eop.mask['mask'] = mask
+
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            tmp_file_name = 'temp_file.tiff'
+            task = ExportToTiff((FeatureType.MASK, 'mask'), folder=tmp_dir_name, date_indices=times)
+            task.execute(eop, filename=tmp_file_name)
+
+            # rasterio saves `bands` to the last dimension, move it up front
+            raster = read_data(os.path.join(tmp_dir_name, tmp_file_name))
+            raster = np.moveaxis(raster, -1, 0)
+
+            self.assertTrue(np.all(subset == raster))
+
+    def test_export2tiff_mask_tuple_int(self):
+        mask = np.arange(5*3*2*1).reshape(5, 3, 2, 1)
+        times = (2, 4)
+        selection = np.arange(times[0], times[1]+1)
+        subset = mask[selection].squeeze()
+
+        eop = EOPatch.load(self.PATCH_FILENAME)
+        eop.mask['mask'] = mask
+
+        with tempfile.TemporaryDirectory() as tmp_dir_name:
+            tmp_file_name = 'temp_file.tiff'
+            task = ExportToTiff((FeatureType.MASK, 'mask'), folder=tmp_dir_name, date_indices=times)
+            task.execute(eop, filename=tmp_file_name)
+
+            # rasterio saves `bands` to the last dimension, move it up front
+            raster = read_data(os.path.join(tmp_dir_name, tmp_file_name))
+            raster = np.moveaxis(raster, -1, 0)
+
+            self.assertTrue(np.all(subset == raster))
+
+    def test_export2tiff_mask_tuple_datetime(self):
+        mask = np.arange(5*3*2*1).reshape(5, 3, 2, 1)
+
+        eop = EOPatch.load(self.PATCH_FILENAME)
+        eop.mask['mask'] = mask
+        dates = np.array(eop.timestamp)

```

Comment:
This is an error, will fix tomorrow, should not really use these timestamps as the newly added features are different... also in other cases